### PR TITLE
56 2.7 application status overview closes #56

### LINKED ISSUE #56: US_2.7: Application Status Overview
**As**
pupil

**I want to**
have an overview over my applications and what their status is

**In order to**
manage my applications more easily

**Dependency:** #12, #45, #18

**Estimate:** 8

**Acceptance criteria**
- The view can be found via an entry in the dropdown in the "user-tab"
- page should look like this
selection_013

- necessary elements:
  - title of event 
  - status ("angenommen", "abgelehnt", "In Bearbeitung" "Beworben")
  - edit-button
- status "Beworben" is shown before the deadline
- edit button is only shown before the deadline
- status "In Bearbeitung" is shown after the deadline
- status "Angenommen/Abgelehnt" are shown if application is accepted/rejected
- by clicking on title user is taken to site of the event

### PR DESCRIPTION
**Pull request by nstrelow**

### CODE REVIEW

**Comment by nstrelow** [comment]
Merge Conflict because of reverting #135 
Please Revert revert or include again, before resolving this merge conflict

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -9,16 +9,13 @@ def dropdown_items
     o = ''
     # everyone gets settings
     o << (menu_item t(:settings, scope: 'navbar'), edit_user_registration_path)
+    o <<  (menu_item t(:my_events, scope: 'navbar'), my_events_path)
```
Comment (cmfcmf) [inline app/helpers/application_helper.rb]: Unnecessary whitespace added behind `<<`

File: app/models/application_letter.rb [hunk 2]
```diff
@@ -19,10 +19,10 @@ class ApplicationLetter < ActiveRecord::Base
   VALID_GRADES = 5..13
 
   validates :user, :event, :experience, :motivation, :coding_skills, :emergency_number, presence: true
-  validates :grade, presence: true, numericality: { only_integer: true }
+  validates :grade, presence: true, numericality: {only_integer: true}
   validates_inclusion_of :grade, :in => VALID_GRADES
-  validates :vegeterian, :vegan, :allergic, inclusion: { in: [true, false] }
-  validates :vegeterian, :vegan, :allergic, exclusion: { in: [nil] }
+  validates :vegeterian, :vegan, :allergic, inclusion: {in: [true, false]}
+  validates :vegeterian, :vegan, :allergic, exclusion: {in: [nil]}
```
Comment (cmfcmf) [inline app/models/application_letter.rb]: I don't think you should remove the whitespace.

File: app/views/my_events/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    <th><%= I18n.t("my_events.index.td_status") %></th>
+    <th><%=t '.actions', :default => t("helpers.actions") %></th>
+  </tr>
+  </thead>
+  <tbody>
+  <% @application_letters.each do |letter|
+    event = Event.find(letter.event_id)
+  %>
+
+      <tr>
```
Comment (cmfcmf) [inline app/views/my_events/index.html.erb]: This looks like it's indented two spaces too wide.

Comment (cmfcmf) [inline app/views/my_events/index.html.erb]: This looks like it's indented two spaces too wide.

File: app/views/my_events/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  <thead>
+  <tr>
+    <th><%= I18n.t("my_events.index.td_events") %></th>
+    <th><%= I18n.t("my_events.index.td_status") %></th>
+    <th><%=t '.actions', :default => t("helpers.actions") %></th>
+  </tr>
+  </thead>
+  <tbody>
+  <% @application_letters.each do |letter|
+    event = Event.find(letter.event_id)
```
Comment (cmfcmf) [inline app/views/my_events/index.html.erb]: This will issue another SQL query for every event.

File: spec/factories/application_letters.rb [hunk 5]
```diff
@@ -37,4 +37,6 @@
   factory :application_letter_rejected, parent: :application_letter do
     status :rejected
   end
+
+
```
Comment (cmfcmf) [inline spec/factories/application_letters.rb]: Can be removed

File: app/controllers/my_events_controller.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class MyEventsController < ApplicationController
+
+  # GET /my_events
+  def index
+    @application_letters = ApplicationLetter.where(user_id: current_user.id)
+  end
+end
```
Comment (cmfcmf) [inline app/controllers/my_events_controller.rb]: I don't think another controller is needed for this.

**Review by cmfcmf — CHANGES_REQUESTED** [review]
I don't think you should create another controller.

**Comment by nstrelow** [comment]
Will start from scratch for that

**Pull request closed without merging**